src/signer.rs

```diff
@@ -109,14 +131,24 @@ where
     fn sign(
         &self,
         extrinsic: SignedPayload<Encoded, E::Extra>,
-    ) -> UncheckedExtrinsic<T::Address, Encoded, S, E::Extra> {
+    ) -> Pin<
+        Box<
+            dyn Future<
+                    Output = Result<
+                        UncheckedExtrinsic<T::Address, Encoded, S, E::Extra>,
+                        String,
+                    >,
+                > + Send
+                + Sync,
+        >,
+    > {
         let signature = extrinsic.using_encoded(|payload| self.signer.sign(payload));
         let (call, extra, _) = extrinsic.deconstruct();
-        UncheckedExtrinsic::new_signed(
+        Box::pin(futures::future::ready(Ok(UncheckedExtrinsic::new_signed(
```

Comment:
```suggestion
        Box::pin(async { Ok(UncheckedExtrinsic::new_signed(
```

Reply:
I tried this and got the following error message:

        Checking substrate-subxt v0.8.0 (/mnt/removable/parity/substrate-subxt)
    error[E0277]: `<P as sp_core::crypto::Pair>::Signature` cannot be shared between threads safely
       --> src/signer.rs:147:9
        |
    144 |       > {
        |        - help: consider further restricting the associated type: `where <P as sp_core::crypto::Pair>::Signature: std::marker::Sync`
    ...
    147 | /         Box::pin(async {
    148 | |             Ok(UncheckedExtrinsic::new_signed(
    149 | |                 call,
    150 | |                 self.account_id.clone().into(),
    ...   |
    153 | |             ))
    154 | |         })
        | |__________^ `<P as sp_core::crypto::Pair>::Signature` cannot be shared between threads safely
        |
        = help: within `impl core::future::future::Future`, the trait `std::marker::Sync` is not implemented for `<P as sp_core::crypto::Pair>::Signature`
        = note: required because it appears within the type `[static generator@src/signer.rs:147:24: 154:10 call:Encoded, self:&&signer::PairSigner<T, S, E, P>, signature:<P as sp_core::crypto::Pair>::Signature, extra:<E as extra::SignedExtra<T>>::Extra _]`
        = note: required because it appears within the type `std::future::GenFuture<[static generator@src/signer.rs:147:24: 154:10 call:Encoded, self:&&signer::PairSigner<T, S, E, P>, signature:<P as sp_core::crypto::Pair>::Signature, extra:<E as extra::SignedExtra<T>>::Extra _]>`
        = note: required because it appears within the type `impl core::future::future::Future`
        = note: required for the cast to the object type `dyn core::future::future::Future<Output = std::result::Result<sp_runtime::generic::unchecked_extrinsic::UncheckedExtrinsic<<T as frame::system::System>::Address, Encoded, S, <E as extra::SignedExtra<T>>::Extra>, std::string::String>> + std::marker::Send + std::marker::Sync`
    
    error[E0277]: `<P as sp_core::crypto::Pair>::Signature` cannot be sent between threads safely
       --> src/signer.rs:147:9
        |
    144 |       > {
        |        - help: consider further restricting the associated type: `where <P as sp_core::crypto::Pair>::Signature: std::marker::Send`
    ...
    147 | /         Box::pin(async {
    148 | |             Ok(UncheckedExtrinsic::new_signed(
    149 | |                 call,
    150 | |                 self.account_id.clone().into(),
    ...   |
    153 | |             ))
    154 | |         })
        | |__________^ `<P as sp_core::crypto::Pair>::Signature` cannot be sent between threads safely
        |
        = help: within `impl core::future::future::Future`, the trait `std::marker::Send` is not implemented for `<P as sp_core::crypto::Pair>::Signature`
        = note: required because it appears within the type `[static generator@src/signer.rs:147:24: 154:10 call:Encoded, self:&&signer::PairSigner<T, S, E, P>, signature:<P as sp_core::crypto::Pair>::Signature, extra:<E as extra::SignedExtra<T>>::Extra _]`
        = note: required because it appears within the type `std::future::GenFuture<[static generator@src/signer.rs:147:24: 154:10 call:Encoded, self:&&signer::PairSigner<T, S, E, P>, signature:<P as sp_core::crypto::Pair>::Signature, extra:<E as extra::SignedExtra<T>>::Extra _]>`
        = note: required because it appears within the type `impl core::future::future::Future`
        = note: required for the cast to the object type `dyn core::future::future::Future<Output = std::result::Result<sp_runtime::generic::unchecked_extrinsic::UncheckedExtrinsic<<T as frame::system::System>::Address, Encoded, S, <E as extra::SignedExtra<T>>::Extra>, std::string::String>> + std::marker::Send + std::marker::Sync`
    
    error[E0277]: `E` cannot be shared between threads safely
       --> src/signer.rs:147:9
        |
    147 | /         Box::pin(async {
    148 | |             Ok(UncheckedExtrinsic::new_signed(
    149 | |                 call,
    150 | |                 self.account_id.clone().into(),
    ...   |
    153 | |             ))
    154 | |         })
        | |__________^ `E` cannot be shared between threads safely
        |
        = help: within `impl core::future::future::Future`, the trait `std::marker::Sync` is not implemented for `E`
    help: consider further restricting this bound with `+ std::marker::Sync`
       --> src/signer.rs:118:5
        |
    118 |     E: SignedExtra<T> + 'static,
        |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^
        = note: required because it appears within the type `(S, E)`
        = note: required because it appears within the type `std::marker::PhantomData<(S, E)>`
        = note: required because it appears within the type `signer::PairSigner<T, S, E, P>`
        = note: required because it appears within the type `&signer::PairSigner<T, S, E, P>`
        = note: required because it appears within the type `&&signer::PairSigner<T, S, E, P>`
        = note: required because it appears within the type `[static generator@src/signer.rs:147:24: 154:10 call:Encoded, self:&&signer::PairSigner<T, S, E, P>, signature:<P as sp_core::crypto::Pair>::Signature, extra:<E as extra::SignedExtra<T>>::Extra _]`
        = note: required because it appears within the type `std::future::GenFuture<[static generator@src/signer.rs:147:24: 154:10 call:Encoded, self:&&signer::PairSigner<T, S, E, P>, signature:<P as sp_core::crypto::Pair>::Signature, extra:<E as extra::SignedExtra<T>>::Extra _]>`
        = note: required because it appears within the type `impl core::future::future::Future`
        = note: required for the cast to the object type `dyn core::future::future::Future<Output = std::result::Result<sp_runtime::generic::unchecked_extrinsic::UncheckedExtrinsic<<T as frame::system::System>::Address, Encoded, S, <E as extra::SignedExtra<T>>::Extra>, std::string::String>> + std::marker::Send + std::marker::Sync`
    
    error: aborting due to 3 previous errors
    
    For more information about this error, try `rustc --explain E0277`.
    error: could not compile `substrate-subxt`.
    
    To learn more, run the command again with --verbose.